carla/models/catalog/catalog.py

```diff
@@ -0,0 +1,260 @@
+        elif ext == "h5":
+            self._backend = "tensorflow"
+        else:
+            raise Exception("Model type not in catalog")
+
+        self._name = model_type + "_" + data_name
+
+        # Preparing pipeline components
+        self._scaler = preprocessing.MinMaxScaler().fit(data.raw[self._continuous])
+
+        if data_name == "adult":
+            self._feature_input_order = [
+                "age",
+                "fnlwgt",
+                "education-num",
+                "capital-gain",
+                "capital-loss",
+                "hours-per-week",
+                "workclass_Private",
+                "marital-status_Non-Married",
+                "occupation_Other",
+                "relationship_Non-Husband",
+                "race_White",
+                "sex_Male",
+                "native-country_US",
+            ]
+            self._encodings = (
+                [  # Encodings should be built in the get_dummy way: {column}_{value}
+                    "workclass_Private",
+                    "marital-status_Non-Married",
+                    "occupation_Other",
+                    "relationship_Non-Husband",
+                    "race_White",
+                    "sex_Male",
+                    "native-country_US",
+                ]
+            )
+        else:
+            raise Exception("Model for dataset not in catalog")
+
+    def pipeline(self, df):
+        """
+        Transforms input for prediction into correct form.
+        Only possible for DataFrames without preprocessing steps.
+
+        Recommended to use to keep correct encodings, normalization and input order
+
+        Parameters
+        ----------
+        df : pd.DataFrame
+            Contains unnormalized and
+
+        Returns
+        -------
+        output : pd.DataFrame
+            Prediction input in correct order, normalized and encoded
+
+        """
+        output = df.copy()
+
+        # Normalization
+        output[self._continuous] = self._scaler.transform(output[self._continuous])
+
+        # Encoding
+        output[self._encodings] = 0
+        for encoding in self._encodings:
+            for cat in self._categoricals:
+                if cat in encoding:
+                    value = encoding.split(cat + "_")[-1]
+                    output.loc[output[cat] == value, encoding] = 1
+                    break
+
+        # Get correct order
+        output = output[self._feature_input_order]
+
+        return output
+
+    def need_pipeline(self, x):
+        """
+        Checks if ML model input needs pipelining.
+        Only DataFrames can be used to pipeline input.
+
+        Parameters
+        ----------
+        x : pd.DataFrame or np.Array
+
+        Returns
+        -------
+        bool : Boolean
+            True if no pipelining process is already taken
+        """
+        if not isinstance(x, pd.DataFrame):
+            return False
+
+        if x.select_dtypes(exclude=[np.number]).empty:
```

Comment:
Don't we also need the pipeline if we have only numbers, which could be still unscaled?

Reply:
Yes, and also if the scaling is wrong. For example standard scaling instead of MinMax scaling, but I'm not sure if it is possible to intercept those cases

Reply:
This condition checks if there are any non-numeric values in the input, if there are, we have to perform encoding on the categorical features.

Reply:
As disscussed, if methods need or pipeline or not could potentially be determined by a flag that the RecourseMethod configures on the MLModel